.cargo-husky/hooks/pre-commit

```diff
@@ -5,6 +5,9 @@
 
 set -e
 
+export RUSTFLAGS=-Dwarnings
```

Comment:
Is my understanding correct that the `RUSTFLAGS` variable will be overwritten after calling `git commit`?

Reply:
The case I was hitting was some code found by CI's `cargo check` to be dead. I tried just adding the `cargo check` line in here, but that wasn't failing the hook because `-Dwarnings` wasn't present yet. With the sum of these changes I was able to replicate the CI result.

No, this script will not overwrite the `RUSTFLAGS` variable in the CLI environment, I specifically double checked that. The `export` here exports it to the environment that's executing this bash script, so that it will apply to all of the commands below, but not to the parent environment. Without `export`, the variable would be set for the context of this bash script itself, but not for its child processes (eg the binaries that the script spawns).

Reply:
Perfect! Thanks for validating 🙂